naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/GraphQLController.scala

```diff
@@ -0,0 +1,92 @@
+package org.coursera.naptime.ari.graphql
+
+import javax.inject._
+
+import com.typesafe.scalalogging.StrictLogging
+import org.coursera.naptime.ari.EngineApi
+import org.coursera.naptime.ari.Response
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+import play.api.libs.json.Json
+import play.api.mvc._
+import sangria.execution.ErrorWithResolver
+import sangria.execution.Executor
+import sangria.execution.QueryAnalysisError
+import sangria.parser.QueryParser
+import sangria.renderer.SchemaRenderer
+import sangria.marshalling.playJson._
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+
+@Singleton
+class GraphQLController @Inject() (
+  graphqlSchemaProvider: GraphqlSchemaProvider,
+  schemaProvider: GraphqlSchemaProvider,
+  engine: EngineApi)
+  (implicit ec: ExecutionContext)
+  extends Controller
+    with StrictLogging {
+
+
+  def index = Action {
+    Ok(SchemaRenderer.renderSchema(graphqlSchemaProvider.schema))
+  }
+
+  def graphqlBody = Action.async(parse.json) { request =>
+    val query = (request.body \ "query").as[String]
+    val operation = (request.body \ "operationName").asOpt[String]
+
+    val variables = (request.body \ "variables").toOption.flatMap {
+      case JsString(vars) => Some(parseVariables(vars))
+      case obj: JsObject => Some(obj)
+      case _ => None
+    }
+
+    executeQuery(query, request, variables, operation)
+      .map(_.withHeaders(("Access-Control-Allow-Origin", "*")))
+  }
+
+  private def parseVariables(variables: String) = {
+    if (variables.trim == "" || variables.trim == "null") {
+      Json.obj()
+    } else {
+      Json.parse(variables).as[JsObject]
+    }
+  }
+
+  private def executeQuery(
+      query: String,
+      requestHeader: RequestHeader,
+      variables: Option[JsObject],
+      operation: Option[String]) = {
+
+    // TODO(bryan): Handle errors / failures properly here
+    (for {
+      request <- SangriaGraphQlParser.parse(query, requestHeader)
+      document <- QueryParser.parse(query).toOption
+    } yield {
+      val fetcherExecution: Future[Option[Response]] = if (query.contains("IntrospectionQuery")) {
```

Comment:
we should add a todo here to just have the engine handle introspection queries naturally instead of having to special case them here. (in those cases, there are just no data needs, so it should return instantly with an empty map of data)